src/wizards/services/WizardService.ts

```diff
@@ -186,12 +192,16 @@ export class WizardService {
     return wizardState.stages[wizardState.indexOfActive];
   }
 
-  private checkCanProceed(indexOfActive: number): boolean {
+  private async checkCanProceed(wizardState: any, indexOfActive: number): Promise<boolean> {
     const visibleStagesCount = this.getVisibleStages().length;
     const isLastStage = visibleStagesCount === indexOfActive + 1;
     /** Proceed just fine when not last stage. */
     if (!isLastStage) return true;
 
+    if (isLastStage) {
+      await this.validateActiveStage(wizardState, indexOfActive );
```

Comment:
@dkent600 
Problem with this:
`validateActiveStage` will be called twice for the Terms Stage
Once: `checkCanProceed` (because proceeding from the "last stage" requires all other stages to be valid, cf. Launch)
Twice: `goToStage`

Imo, negligible.
(if not, I would have to refactor `goToStage` a bit, as I see it)

Reply:
I suspect we can live with that.